nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -95,10 +96,19 @@ impl<'a> Planner<'a> {
     }
 
     pub fn plan(mut self) -> Result<Blueprint, Error> {
+        self.check_input_validity()?;
         self.do_plan()?;
         Ok(self.blueprint.build())
     }
 
+    fn check_input_validity(&self) -> Result<(), Error> {
```

Comment:
I much prefer this to taking the parameter into the allocator. Thanks for doing this!